packages/flutter/lib/src/rendering/sliver_fill.dart

```diff
@@ -124,7 +135,7 @@ class RenderSliverFillRemaining extends RenderSliverSingleBoxAdapter {
     assert(paintedChildSize.isFinite);
     assert(paintedChildSize >= 0.0);
     geometry = SliverGeometry(
-      scrollExtent: constraints.viewportMainAxisExtent,
+      scrollExtent: hasScrollBody ? constraints.viewportMainAxisExtent : constraints.scrollOffset,
```

Comment:
(GitHub is not letting me comment on the old thread anymore)

I still think that constraints.scrollOffset is coming in as zero here. Maybe you can show in a test that the scrollExtent of the RenderSLiverFillRemaining is actually set to the expected non-zero value?

Reply:
Do you mean that instead of `constraints.scrollOffset`, I may be able to just use `0`?

Reply:
Let's look at this in-person when you have a minute.